ReactSkia/sdk/WindowDelegator.cpp

```diff
@@ -61,16 +61,34 @@ void  WindowDelegator::createNativeWindow() {
 }
 
 void WindowDelegator::closeWindow() {
-  RNS_LOG_TODO("Sync between rendering & Exit to be handled ");
   windowActive = false;
-  std::scoped_lock lock(renderCtrlMutex);
 
   if(ownsTaskrunner_) {
+    windowTaskRunner_->dispatch([&](){
+      closeNativeWindow();
+      sem_post(&semReadyToDraw_);
+    });
+
+    sem_wait(&semReadyToDraw_);
     windowTaskRunner_->stop();
+
+  } else {
+    closeNativeWindow();
   }
+
   if (workerThread_.joinable() ) {
     workerThread_.join();
   }
+  sem_destroy(&semReadyToDraw_);
+  windowDelegatorCanvas=nullptr;
+  windowReadyTodrawCB_=nullptr;
+
+}
+
+void WindowDelegator::closeNativeWindow() {
+  RNS_LOG_TODO("Sync between rendering & Exit to be handled ");
```

Comment:
can this, TO Do  be removed. It was missed in last commit. Synchronization was already achieved through mutex

Reply:
updated